log_enricher/__init__.py

```diff
@@ -5,39 +5,47 @@
 import sys
 import threading
 
+from typing import Callable, Dict, List, Optional
+
+from .enrichers import Enricher, ConfigProperty, Host, Thread, Timestamp
+
 
 class ContextFilter(logging.Filter):
-    def __init__(self, config, request_id_getter, user_context_getter):
+    def __init__(self, enrichers: List[Callable] = None):
         super().__init__()
-        self.config = config
-        self.request_id_getter = request_id_getter
-        self.user_context_getter = user_context_getter
+        if enrichers is None:
+            enrichers = []
+        self._enrichers = enrichers
 
     def filter(self, record):
-        record.app_version = self.config.get("app_version")
-        record.release_stage = self.config.get("release_stage")
-        record.host = platform.node()
-        record.thread_id = threading.current_thread().getName()
-        if self.request_id_getter:
-            request_id = self.request_id_getter()
-            if request_id:
-                record.request_id = request_id
-        if self.user_context_getter:
-            user_context = self.user_context_getter()
-            if user_context:
-                record.username = user_context.get("username")
-                record.user_id = user_context.get("user_id")
-                record.path = user_context.get("path")
-        record.timestamp = datetime.datetime.now().isoformat(sep="T", timespec="milliseconds")
+        for enricher in self._enrichers:
+            props = enricher()
+            for attr, value in props.items():
+                setattr(record, attr, value)
+
+        # XXX: replace these with enrichers which take in record as an argument
```

Comment:
We should dig into why these were necessary (datadog?) and at least capture that context so when they're refactored it can take into account their purpose.

Reply:
Python Json Logger skips natural LogRecord attributes
https://github.com/madzak/python-json-logger/blob/master/src/pythonjsonlogger/jsonlogger.py#L16-L22